runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/GrpcDispatcherClientTest.java

```diff
@@ -0,0 +1,159 @@
+          PipelineOptionsFactory.as(DataflowWorkerHarnessOptions.class);
+      options.setExperiments(
+          Lists.newArrayList(
+              GrpcDispatcherClient.STREAMING_ENGINE_USE_JOB_SETTINGS_FOR_ISOLATED_CHANNELS));
+      GrpcDispatcherClient dispatcherClient =
+          GrpcDispatcherClient.create(options, new FakeWindmillStubFactoryFactory(stubFactory));
+      // Create first time with Isolated channels disabled
+      dispatcherClient.onJobConfig(getGlobalConfig(/*useWindmillIsolatedChannels=*/ false));
+      CloudWindmillServiceV1Alpha1Stub stub1 = dispatcherClient.getWindmillServiceStub();
+      CloudWindmillServiceV1Alpha1Stub stub2 = dispatcherClient.getWindmillServiceStub();
+      assertSame(stub2, stub1);
+
+      // Enable Isolated channels
+      dispatcherClient.onJobConfig(getGlobalConfig(/*useWindmillIsolatedChannels=*/ true));
+      CloudWindmillServiceV1Alpha1Stub stub3 = dispatcherClient.getWindmillServiceStub();
+      assertNotSame(stub3, stub1);
+
+      CloudWindmillServiceV1Alpha1Stub stub4 = dispatcherClient.getWindmillServiceStub();
+      assertSame(stub3, stub4);
+
+      // Disable Isolated channels
+      dispatcherClient.onJobConfig(getGlobalConfig(/*useWindmillIsolatedChannels=*/ false));
+      CloudWindmillServiceV1Alpha1Stub stub5 = dispatcherClient.getWindmillServiceStub();
+      assertNotSame(stub4, stub5);
+    }
+  }
+
+  @RunWith(Parameterized.class)
+  public static class RespectsPipelineOptionsTest extends GrpcDispatcherClientTestBase {
+
+    @Parameters
+    public static Collection<Object[]> data() {
+      List<Object[]> list = new ArrayList<>();
+      for (Boolean pipelineOption : new Boolean[] {null, true, false}) {
+        list.add(new Object[] {/*experimentEnabled=*/ false, pipelineOption});
+      }
+      for (Boolean pipelineOption : new Boolean[] {true, false}) {
+        list.add(new Object[] {/*experimentEnabled=*/ true, pipelineOption});
+      }
+      return list;
+    }
+
+    @Parameter(0)
+    public Boolean experimentEnabled;
+
+    @Parameter(1)
+    public Boolean pipelineOption;
+
+    @Test
+    public void ignoresIsolatedChannelsConfigWithPipelineOption() {
+      DataflowWorkerHarnessOptions options =
+          PipelineOptionsFactory.as(DataflowWorkerHarnessOptions.class);
+      if (experimentEnabled) {
+        options.setExperiments(
+            Lists.newArrayList(
+                GrpcDispatcherClient.STREAMING_ENGINE_USE_JOB_SETTINGS_FOR_ISOLATED_CHANNELS));
+      }
+      options.setUseWindmillIsolatedChannels(pipelineOption);
+      GrpcDispatcherClient dispatcherClient =
+          GrpcDispatcherClient.create(options, new FakeWindmillStubFactoryFactory(stubFactory));
+
+      // Job setting disabled, PipelineOption enabled
+      dispatcherClient.onJobConfig(getGlobalConfig(/*useWindmillIsolatedChannels=*/ false));
+      CloudWindmillServiceV1Alpha1Stub stub1 = dispatcherClient.getWindmillServiceStub();
```

Comment:
verify the stub is using isolation matching pipeline option?

Reply:
Done